core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerTwoFacade.java

```diff
@@ -4,64 +4,70 @@
 import android.net.Uri;
 import android.os.Handler;
 import android.os.Looper;
-import android.view.Surface;
 
-import com.google.android.exoplayer2.C;
 import com.google.android.exoplayer2.DefaultLoadControl;
-import com.google.android.exoplayer2.ExoPlaybackException;
-import com.google.android.exoplayer2.ExoPlayer;
 import com.google.android.exoplayer2.ExoPlayerFactory;
-import com.google.android.exoplayer2.Format;
-import com.google.android.exoplayer2.PlaybackParameters;
 import com.google.android.exoplayer2.SimpleExoPlayer;
-import com.google.android.exoplayer2.Timeline;
-import com.google.android.exoplayer2.decoder.DecoderCounters;
-import com.google.android.exoplayer2.source.AdaptiveMediaSourceEventListener;
-import com.google.android.exoplayer2.source.ExtractorMediaSource;
 import com.google.android.exoplayer2.source.MediaSource;
-import com.google.android.exoplayer2.source.TrackGroupArray;
 import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
-import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
 import com.google.android.exoplayer2.ui.SimpleExoPlayerView;
-import com.google.android.exoplayer2.upstream.DataSpec;
 import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
-import com.google.android.exoplayer2.video.VideoRendererEventListener;
 import com.novoda.noplayer.ContentType;
-import com.novoda.noplayer.listeners.BitrateChangedListeners;
-import com.novoda.noplayer.listeners.InfoListeners;
 
-import java.io.IOException;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 
-public class ExoPlayerTwoFacade implements VideoRendererEventListener {
+public class ExoPlayerTwoFacade {
 
     private static final boolean RESET_POSITION = true;
     private static final boolean RESET_STATE = false;
 
-    private int videoWidth;
-    private int videoHeight;
-
     private final SimpleExoPlayer exoPlayer;
     private final MediaSourceFactory mediaSourceFactory;
-
-    private List<Listener> listeners = new CopyOnWriteArrayList<>();
-    private InfoListeners infoListeners;
-    private BitrateChangedListeners bitrateChangedListeners;
-    private InternalErrorListener internalErrorListener;
+    private final EventListener exoPlayerEventListener;
+    private final MediaSourceEventListener mediaSourceEventListener;
```

Comment:
Extract all of the different "Event" Listeners in `ExoPlayer` into separate classes passing each a reference to `List<Forwarders>`, personally I don't like that we can make state changes on `List<Forwarders>` here and it will affect all of the other listeners but we couldn't come up with a nice way of doing this because of the creation order.